plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/ArtifactHandler.kt

```diff
@@ -111,9 +113,11 @@ class ArtifactHandler(private val project: Project, private val clientAdaptor: G
 
     /**
      * Downloads an artifact and returns a [DownloadArtifactResult]
-     * [DownloadArtifactResult.DownloadFailure] indicates failure when downloading artifact
-     * [DownloadArtifactResult.Failure] indicates failure when processing artifact
      * [DownloadArtifactResult.Success] indicates success when downloading and processing artifact
+     * [DownloadArtifactResult.DownloadFailure] indicates failure when downloading artifact
+     * [DownloadArtifactResult.ParseZipFailure] indicates failure when parsing artifact contents
+     * [DownloadArtifactResult.UnzipFailure] indicates failure when unzipping artifact contents to disk
```

Comment:
I think ParseZip and Unzip are the same meaning - maybe ParseArtifact or ApplyPatch can represent issues with applying ungraded code to original codebase better?

Reply:
Not really, it could help us distinguish things like HDD access issues from logical errors in the files themselves. Rn the actual handling is the same for both though, so added it more as a future thing :D